src/org/zalando/stups/mint/worker/job/run.clj

```diff
@@ -22,17 +23,23 @@
   (try
     (when (time/after? (time/now)
                        @rate-limited-until)
-      (let [storage-url (config/require-config configuration :mint-storage-url)]
+      (let [storage-url (config/require-config configuration :mint-storage-url)
+            kio-url (config/require-config configuration :kio-url)]
         (log/debug "Starting new synchronisation run with %s..." configuration)
 
-        (let [apps (storage/list-apps storage-url tokens)]
-          (log/debug "Found apps: %s ..." apps)
-          (doseq [app apps]
+        (let [mint-apps (storage/list-apps storage-url tokens)
+              kio-apps (apps/list-apps kio-url tokens)]
```

Comment:
I would add another line with "kio-apps-by-id" (or similar) where the Kio app can be looked up by its ID